pkg/cmd/server/kubernetes/master_config_test.go

```diff
@@ -1,14 +1,130 @@
 package kubernetes
 
 import (
+	"net"
 	"reflect"
 	"testing"
+	"time"
 
+	apiserveroptions "k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	cmapp "k8s.io/kubernetes/cmd/kube-controller-manager/app/options"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
 	"k8s.io/kubernetes/pkg/genericapiserver"
+	kubeletclient "k8s.io/kubernetes/pkg/kubelet/client"
+	etcdstorage "k8s.io/kubernetes/pkg/storage/etcd"
+	"k8s.io/kubernetes/pkg/util"
 
 	configapi "github.com/openshift/origin/pkg/cmd/server/api"
 )
 
+func TestAPIServerDefaults(t *testing.T) {
+	defaults := apiserveroptions.NewAPIServer()
+
+	// This is a snapshot of the default config
+	// If the default changes (new fields are added, or default values change), we want to know
+	// Once we've reacted to the changes appropriately in BuildKubernetesMasterConfig(), update this expected default to match the new upstream defaults
+	expectedDefaults := &apiserveroptions.APIServer{
+		ServerRunOptions: &genericapiserver.ServerRunOptions{
+			BindAddress:          net.ParseIP("0.0.0.0"),
+			CertDirectory:        "/var/run/kubernetes",
+			InsecureBindAddress:  net.ParseIP("127.0.0.1"),
+			InsecurePort:         8080,
+			LongRunningRequestRE: "(/|^)((watch|proxy)(/|$)|(logs?|portforward|exec|attach)/?$)",
+			SecurePort:           6443,
+		},
+		APIGroupPrefix:          "/apis",
+		APIPrefix:               "/api",
+		AdmissionControl:        "AlwaysAdmit",
+		AuthorizationMode:       "AlwaysAllow",
+		DeleteCollectionWorkers: 1,
+		EnableLogsSupport:       true,
+		EtcdConfig: etcdstorage.EtcdConfig{
+			Prefix: "/registry",
+		},
+		EventTTL:               1 * time.Hour,
+		MasterCount:            1,
+		MasterServiceNamespace: "default",
+		RuntimeConfig:          util.ConfigurationMap{},
+		StorageVersions:        registered.AllPreferredGroupVersions(),
+		DefaultStorageVersions: registered.AllPreferredGroupVersions(),
+		KubeletConfig: kubeletclient.KubeletClientConfig{
+			Port:        10250,
+			EnableHttps: true,
+			HTTPTimeout: time.Duration(5) * time.Second,
+		},
+	}
+
+	if !reflect.DeepEqual(defaults, expectedDefaults) {
```

Comment:
It doesn't seem likely that this will do what you want it to do.  We want to know when they change defaults, but we also want to know when they add fields which suddenly give us new behavior right?  As a for instance, they could just have easily added `DisableLogsSupport` (they've done negative logic fields before) and we'd be in the same boat.

If we're going to make change explicitly opt-in, let's make it explicitly opt in.  You could do it with default serialization to JSON right?  That would be stable and alert us to changes.